### PR TITLE
[JENKINS-63016] feat: new resource - ephemeral storage

### PR DESCRIPTION
**Pull request by KrapivinAndrey**
https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#setting-requests-and-limits-for-local-ephemeral-storage

### CODE REVIEW

**Comment by KrapivinAndrey** [comment]
@Vlatombe

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java [hunk 1]
```diff
@@ -500,10 +500,11 @@ private VolumeMount getDefaultVolumeMount(@CheckForNull String workingDir) {
         return commands;
     }
 
-    private Map<String, Quantity> getResourcesMap(String memory, String cpu) {
+    private Map<String, Quantity> getResourcesMap(String memory, String cpu, String ephemeral) {
         ImmutableMap.Builder<String, Quantity> builder = ImmutableMap.<String, Quantity>builder();
         String actualMemory = substituteEnv(memory);
         String actualCpu = substituteEnv(cpu);
+        String actualEpemeral = substituteEnv(ephemeral);
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java:507]:
```suggestion
        String actualEphemeral = substituteEnv(ephemeral);
```

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java [hunk 2]
```diff
@@ -512,6 +513,10 @@ private VolumeMount getDefaultVolumeMount(@CheckForNull String workingDir) {
             Quantity cpuQuantity = new Quantity(actualCpu);
             builder.put("cpu", cpuQuantity);
         }
+        if (StringUtils.isNotBlank(actualEpemeral)) {
+            Quantity ephemeralQuantity = new Quantity(actualEpemeral);
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java:517]:
```suggestion
            Quantity ephemeralQuantity = new Quantity(actualEphemeral);
```

**Review by Vlatombe — COMMENTED** [review]
Looks good to me overlal. Could you rename various references to `ephemeralStorage` to clarify? Otherwise it feels a little confusing

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java (hunk 1 above)
Comment (KrapivinAndrey) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java]: fix

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java (hunk 2 above)
Comment (KrapivinAndrey) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java]: fix

**Comment by jglick** [comment]
Should we really be adding more and more GUI for this sort of thing? My impression was that we were deprecating specific fields in `PodTemplate` & `ContainerTemplate` in favor of YAML.

**Comment by KrapivinAndrey** [comment]
> GUI

I do not know. I don't use the GUI at all

**Comment by jglick** [comment]
Sorry, I should not have said GUI exactly. These controls _do_ appear in the GUI, in both cloud config screens for global pod template definitions and in **Pipeline Syntax** for the `podTemplate` step; but equally apply to Groovy syntax for this step. The point is that `ephemeral-storage`, along with dozens of other detailed options for pods, can already be specified in YAML without any special support in Jenkins.

**Comment by KrapivinAndrey** [comment]
> Sorry, I should not have said GUI exactly. These controls _do_ appear in the GUI, in both cloud config screens for global pod template definitions and in **Pipeline Syntax** for the `podTemplate` step; but equally apply to Groovy syntax for this step. The point is that `ephemeral-storage`, along with dozens of other detailed options for pods, can already be specified in YAML without any special support in Jenkins.

I understand. But if there is a section for resources, it's strange to configure other resources via yaml

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 3]
```diff
@@ -662,6 +666,18 @@ public void setResourceLimitCpu(String resourceLimitCpu) {
...
+    @Deprecated
+    @DataBoundSetter
+    public void setResourceLimitEphemeral(String resourceLimitEphemeral) {
+        getFirstContainer().ifPresent((i) -> i.setResourceLimitEphemeral(resourceLimitEphemeral));
+    }
+
+    @Deprecated
+    public String getResourceLimitEphemeral() {
+        return getFirstContainer().map(ContainerTemplate::getResourceLimitEphemeral).orElse(null);
+    }
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java:678]: Remove these.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 4]
```diff
@@ -684,6 +700,17 @@ public void setResourceRequestCpu(String resourceRequestCpu) {
...
+    @Deprecated
+    @DataBoundSetter
+    public void setResourceRequestEphemeral(String resourceRequestEphemeral) {
+        getFirstContainer().ifPresent((i) -> i.setResourceRequestEphemeral(resourceRequestEphemeral));
+    }
+    
+    @Deprecated
+    public String getResourceRequestEphemeral() {
+        return getFirstContainer().map(ContainerTemplate::getResourceRequestEphemeral).orElse(null);
+    }
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java:712]: Remove these.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java [hunk 5]
```diff
@@ -57,9 +57,13 @@
 
     private String resourceRequestMemory;
 
+    private String resourceRequestEphemeral;
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java:60]: Replace `ephemeral` with `ephemeralStorage` in the PR please.

**Review by Vlatombe — CHANGES_REQUESTED** [review]
Needs a bit more work

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java (hunk 5 above)
Comment (KrapivinAndrey) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java]: Fix

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java (hunk 4 above)
Comment (KrapivinAndrey) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Fix

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java (hunk 3 above)
Comment (KrapivinAndrey) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Fix

**Review by Vlatombe — APPROVED** [review]

**Pull request merged**